library/alloc/src/collections/vec_deque/tests.rs

```diff
@@ -210,6 +210,20 @@ fn make_contiguous_small_free() {
     );
 }
 
+#[test]
+fn make_contiguous_head_to_end() {
+    let mut dq = VecDeque::with_capacity(3);
+    dq.push_front('B');
+    dq.push_front('A');
+    dq.push_back('C');
+    dq.make_contiguous();
+    let expected_tail = 0;
+    let expected_head = 3;
+    assert_eq!(expected_tail, dq.tail);
+    assert_eq!(expected_head, dq.head);
+    assert_eq!((&['A', 'B', 'C'] as &[_], &[] as &[_]), dq.as_slices());
+}
```

Comment:
Do you want to write a version of this that uses `Box`es so that we ensure it doesn't segfault?

Reply:
that shouldn't matter i think. The reason this could segfault is if the result is wrong. But as the used algorithm doesn't depend on `T` asserting that it is correct for `i32` also means that it would be correct for `Box` and therefore can't segfault.

Reply:
Probably, but I figured better safe than sorry :)

Reply:
yeah, don't have the capacity for this rn though.